test/e2e/tests/request-queuing/dapp1-switch-dapp2-send.spec.js

```diff
@@ -125,28 +116,26 @@ describe('Request Queuing Dapp 1, Switch Tx -> Dapp 2 Send Tx', function () {
 
         await driver.clickElement('#sendButton');
 
-        await switchToNotificationWindow(driver, 4);
-        await driver.findClickableElements({
-          text: 'Switch network',
-          tag: 'button',
-        });
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        const windowsBefore = await driver.getAllWindowHandles();
 
         await driver.clickElement({ text: 'Switch network', tag: 'button' });
 
         // Wait for switch confirmation to close then tx confirmation to show.
-        await driver.waitUntilXWindowHandles(3);
-        await driver.delay(regularDelayMs);
+        const newDialogWindow =
+          await driver.getNewNotificationHandleAfterOldOneCloses({
+            driver,
+            windowsBefore,
+          });
```

Comment:
here is where it happened the race condition, making the test fail. We now use the new driver function to mitigate this